src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/DetailedReportService.java

```diff
@@ -0,0 +1,125 @@
+import org.slf4j.Logger;
+import org.springframework.stereotype.Service;
+import org.springframework.util.Assert;
+import uk.gov.hmcts.reform.blobrouter.clients.bulkscanprocessor.BulkScanProcessorClient;
+import uk.gov.hmcts.reform.blobrouter.config.TargetStorageAccount;
+import uk.gov.hmcts.reform.blobrouter.data.reconciliation.reports.ReconciliationReportRepository;
+import uk.gov.hmcts.reform.blobrouter.data.reconciliation.reports.model.ReconciliationReport;
+import uk.gov.hmcts.reform.blobrouter.data.reconciliation.statements.model.EnvelopeSupplierStatement;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.report.ReconciliationMapper;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.report.ReconciliationReportResponse;
+import uk.gov.hmcts.reform.blobrouter.reconciliation.report.ReconciliationStatement;
+
+import java.time.LocalDate;
+import java.util.Objects;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+@Service
+public class DetailedReportService {
+
+    private static final Logger logger = getLogger(DetailedReportService.class);
+
+    private final ReconciliationMapper reconciliationMapper;
+    private final ReconciliationReportRepository reconciliationReportRepository;
+    private final ReconciliationService reconciliationService;
+    private final BulkScanProcessorClient bulkScanProcessorClient;
+    private final ObjectMapper objectMapper;
+
+    public DetailedReportService(
+        ReconciliationMapper reconciliationMapper,
+        ReconciliationReportRepository reconciliationReportRepository,
+        ReconciliationService reconciliationService,
+        BulkScanProcessorClient bulkScanProcessorClient,
+        ObjectMapper objectMapper
+    ) {
+        this.reconciliationMapper = reconciliationMapper;
+        this.reconciliationReportRepository = reconciliationReportRepository;
+        this.reconciliationService = reconciliationService;
+        this.bulkScanProcessorClient = bulkScanProcessorClient;
+        this.objectMapper = objectMapper;
+    }
+
+    public void process(LocalDate date, TargetStorageAccount account) {
+
+        Assert.isTrue(
+            account == TargetStorageAccount.BULKSCAN,
+            "Only BULKSCAN account can be processed."
+        );
+
+        var optionReconciliationReport =
+            reconciliationReportRepository.getLatestReconciliationReport(date, account.name());
+
+        if (optionReconciliationReport.isPresent()) {
+            var reconciliationReport = optionReconciliationReport.get();
+
+            if (Objects.nonNull(reconciliationReport.detailedContent)) {
+                logger.info(
+                    "Reconciliation detailed report already processed."
+                        + "Supplier Statement Id: {}, Report id: {}, Created at: {}",
+                    reconciliationReport.supplierStatementId,
+                    reconciliationReport.id,
+                    reconciliationReport.createdAt
+                );
+                return;
```

Comment:
maybe we can exclude those in SQL query?

Reply:
as there are not many columns,  I think it is fine. And it is good for logging.